src/include/shared/sgxlkl_enclave_config_gen.h

```diff
@@ -0,0 +1,148 @@
+#ifndef SGXLKL_ENCLAVE_CONFIG_GEN_H
+#define SGXLKL_ENCLAVE_CONFIG_GEN_H
+
+/* Automatically generated from ../../tools/schemas/enclave-config.schema.json; do not modify. */
+
+#include <inttypes.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <elf.h>
+
+#define SGXLKL_ENCLAVE_CONFIG_VERSION 1UL
+
+typedef enum
+{
+    UNKNOWN_MODE = 0,
+    SW_DEBUG_MODE = 1,
+    HW_DEBUG_MODE = 2,
+    HW_RELEASE_MODE = 3
+} sgxlkl_enclave_mode_t;
+
+const char* sgxlkl_enclave_mode_t_to_string(sgxlkl_enclave_mode_t e);
+sgxlkl_enclave_mode_t string_to_sgxlkl_enclave_mode_t(const char *e);
+
+typedef enum
+{
+    ENCLAVE_MMAP_FILES_NONE = 0,
+    ENCLAVE_MMAP_FILES_PRIVATE = 1,
+    ENCLAVE_MMAP_FILES_SHARED = 2
+} sgxlkl_enclave_mmap_files_t;
+
+const char* sgxlkl_enclave_mmap_files_t_to_string(sgxlkl_enclave_mmap_files_t e);
+sgxlkl_enclave_mmap_files_t string_to_sgxlkl_enclave_mmap_files_t(const char *e);
+
+typedef enum
+{
+    EXIT_STATUS_FULL = 0,
+    EXIT_STATUS_BINARY = 1,
+    EXIT_STATUS_NONE = 2
+} sgxlkl_exit_status_mode_t;
+
+const char* sgxlkl_exit_status_mode_t_to_string(sgxlkl_exit_status_mode_t e);
+sgxlkl_exit_status_mode_t string_to_sgxlkl_exit_status_mode_t(const char *e);
+
+typedef struct sgxlkl_clock_res_config
+{
+    char resolution[17];
+} sgxlkl_clock_res_config_t;
```

Comment:
Is this dead code?

Reply:
No, but I'm somewhat hopeful that it will soon be dead. @prp and I think we need only one of them, but neither of us is sure. @SeanTAllen do you understand which parts of the clock resolutions we still need and whether they need to be in the attested config? We thought perhaps we need only one of them, and it could be computed in the enclave instead of trusting the host about it.

Reply:
@SeanTAllen @prp @wintersteiger Please work together to figure this out :)

Reply:
I'll open a separate issue for this; we don't have to resolve that right here and now.

Reply:
@wintersteiger I do not. The clock resolution didn't have anything to do with the time related work I did which was swapping out the lower level nano source from one source of nanos to another.